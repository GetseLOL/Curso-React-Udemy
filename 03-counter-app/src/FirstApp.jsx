import React from "react";
import PropTypes from "prop-types";
import "./FirstApp.css";

const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subTitle}</h2>
      <h2>{subTitle}</h2>
      <h3>{name}</h3>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  // title: "No hay titulo",
  subTitle: "No hay subtitulo",
  name: "No hay titulo",
};

export default FirstApp;
