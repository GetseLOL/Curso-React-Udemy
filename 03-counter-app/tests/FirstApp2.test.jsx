import { screen, render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
  const title = "Hola soy goku";
  const subTitle = "Soy un subtitulo"; 
  test("debe hacer natchcon el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar el mensaje hola soy goku", () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe mostrar el titulo en h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);

    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
