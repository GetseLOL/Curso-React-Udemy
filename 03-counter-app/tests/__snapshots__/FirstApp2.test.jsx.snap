// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer natchcon el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <h2>
    No hay subtitulo
  </h2>
  <h2>
    No hay subtitulo
  </h2>
  <h3>
    No hay titulo
  </h3>
</div>
`;
